### DISCUSSION TITLE
Shortcut Improvements

### DISCUSSION THREAD

**Discussion by OverloadedOrama** [Orama-Interactive/Pixelorama#588] [Ideas]
**Feature/enhancement description:**

- [x] Solve the macOS hotkey issue (#67).
- [x] Holding Ctrl+Z to undo many changes quickly. (#405)
- [x] Add the ability to change shortcut bindings. Currently only implemented for tools and color switch, but it would be nice to be able to modify all shortcuts. Perhaps we should make it that the user can modify the entire (or most) of the Input Map bindings found in the project settings. This should also include the shortcuts used for each tool, for example Pencil and Shift to draw lines. (#700)
- [x] Clear shortcut button. Not all shortcuts have to have a binding, the ability to clear a shortcut binding would be handy. (#700)
- [x] Let the users create their own presets. I'm not sure how useful this is, but since we already use a Preset system, we might as well let the user save their changes as a preset and easily switch between them, like different "profiles". **EDIT**: Tracked in https://github.com/Orama-Interactive/Keychain/discussions/3
- [ ] Add missing shortcuts for manipulating layers, frames, palettes and whatever else is missing. I might edit this to add a proper list of the missing shortcuts, feel free to give ideas! #900

**Comment by Blixtmen** [comment]
Being able to hold ctrl+z to undo many changes quickly is something that is missed from other softwares!

**Comment by OverloadedOrama** [comment]
Currently waiting for https://github.com/godotengine/godot/issues/63108 to be resolved in Godot, so the missing buttons can have shortcuts. I have already opened a PR to fix this issue, hopefully it will get merged soon.
